components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -987,6 +998,7 @@ subroutine p3_main_part3(kts, kte, kbot, ktop, kdir, &
 
       ! ice:
 
+      !PMC potential bug: I think *all* of this qi stuff needs to be incld!!!
       call impose_max_total_Ni(ni(k),max_total_Ni,inv_rho(k))
```

Comment:
@hassanbeydoun and @JS-WRF-SBM - I think all of this ice-phase stuff (from this comment to the end of p3_main_part3) should be in-cloud calculations. I haven't made the change because it requires passing cld_frac_i into p3_main_part3, which is a huge hassle in our C++ framework. Do you agree that this is needed? Kobby - did you already implement this in your version? My argument here is that qi distributions and resulting table lookups only make sense in an in-cloud format (and in fact, p3 and its table lookup was originally built to only be called when cloud fraction was 100%).